network.py

```diff
@@ -0,0 +1,23 @@
+import numpy as np
+import math
+import random
+from neuron import Neuron, sig
+from layer import Layer, sep
+
+class Network:
+    def __init__(self, archSpec):
+        self.archSpec = archSpec
+        self.input = [0]*archSpec[0] #initialize empty input layer array
```

Comment:
My thought was to pre-allocate memory of the dimension of our input layer because my understanding is that vector operations are parallelized in Python thus making them run faster, whereas if we instantiated a list object without this pre-allocated memory then I would have to essentially iterate within a for loop to append values from our input (which I think is suboptimal).

I have no clue if this is actually the case, or if there is a vector operation version of appending values to a list, but I'll do some more research on it and see what I can find.

Reply:
Oh nice I just learned about vector operations. This is cool. Keep going with this. :)